src/http/client.ts

```diff
@@ -186,13 +216,20 @@ export class HttpClient {
    * @param request - The HTTP request object to which headers are added.
    */
   private attachHeaders(request: Request) {
-    // Set the content-type header to JSON
-    request.headers.set('Content-Type', 'application/json');
+    this.setContentTypeHeader(request);
 
     // Set auth headers if available, potentially overwrite manually set auth headers
     this._authManager.authenticateRequest(request);
   }
 
+  private setContentTypeHeader(request: Request) {
+    const [key, value] = ['Content-Type', 'application/json'];
+
+    request.headers.set(key, value);
+
+    debug('%o header set to %o for %o', key, value, RequestHelper.format(request));
+  }
```

Comment:
Do we need this? Could we instead have a setHeader(key, value)?

Reply:
Where would the log originate from? `RequestHelper.setHeader(req, name, value)` or from the http client when it calls it?

Let's continue the discussion and possibly add this in a later PR